### PR TITLE
doc: added section on creating RESTful API user

### PR DESCRIPTION
**Pull request by falcon78921**
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

This will help users better understand how to
add RESTful API users/keys.

Fixes: https://tracker.ceph.com/issues/37957
Signed-off-by: James McClune <jmcclune@mcclunetechnologies.net>

### CODE REVIEW

**Comment by falcon78921** [comment]
jenkins retest please

**Comment by falcon78921** [comment]
jenkins retest this please

**Comment by falcon78921** [comment]
jenkins retest this please

**Comment by falcon78921** [comment]
See: http://docs.ceph.com/ceph-prs/26016/mgr/restful/#creating-an-api-user

**Comment by falcon78921** [comment]
Hi @gregsfortytwo, I saw you labeled this issue awhile back. When you have a chance, could you please review? :)

**Comment by gregsfortytwo** [comment]
Sorry, I'm just scrubbing so it shows up in the manager queue and they can check it. :)

File: doc/mgr/restful.rst [hunk 1]
```diff
@@ -77,6 +77,103 @@ If the port is not configured, *restful* will bind to port ``8003``.
...
+-----------------------
+
+To create an API user, please run the following command::
+
+  ceph restful create-key <username>
+
+Replace ``<username>`` with the desired name of the user. For example, to create a user named
+``api``::
+
+  root@system1:/home# ceph restful create-key api
```
Comment (tchaikov) [inline doc/mgr/restful.rst]: could you remove the prompt so the command line example look like
```
$  ceph restful create-key api
52dffd92-a103-4a10-bfce-5b60f48f764e
```

File: doc/mgr/restful.rst [hunk 2]
```diff
@@ -77,6 +77,103 @@ If the port is not configured, *restful* will bind to port ``8003``.
...
+
+The ``ceph restful list-keys`` command will output in JSON::
+
+  {
+  	"api": "52dffd92-a103-4a10-bfce-5b60f48f764e"
+  }
+
+You can use ``curl`` in order to test your user with the API. Here is an example::
+
+  curl -k https://api:52dffd92-a103-4a10-bfce-5b60f48f764e@192.168.5.10:8003/server
```
Comment (tchaikov) [inline doc/mgr/restful.rst]: please note, we are using `<ceph-mgr>:<port>` in the rest of this document. and i think we can skip the output of the `curl` command here.

File: doc/mgr/restful.rst (hunk 1 above)
Comment (falcon78921) [reply doc/mgr/restful.rst]: @tchaikov Sure :)

**Comment by falcon78921** [comment]
jenkins render docs

**Comment by falcon78921** [comment]
See: http://docs.ceph.com/ceph-prs/26016/mgr/restful/#creating-an-api-user

**Comment by falcon78921** [comment]
@tchaikov I'm just following up on this PR. I made the requested changes. If there's anything else, please let me know.

**Comment by tchaikov** [comment]
thank you @falcon78921 ! sorry for the latency.

**Comment by falcon78921** [comment]
@tchaikov No prob! :)

**Pull request merged**